v2/codetags/extractor.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package codetags
+
+import (
+	"strings"
+)
+
+// Extract extracts comments for special metadata tags. Lines are filtered by a
+// string prefix that matching lines must start with.
+//
+// Example: When called with prefix "+k8s:", lines:
+//
+//	Comment line without marker
+//	+k8s:required
```

Comment:
Can you include a (fake) example which explains why it returns `[]string` ?

Reply:
Also mayber an example with args to show why the map-key is included in the value?  E.g. I think it would be somewthing like:

```
"withArgs": {"withArgs(abc123)=value"},
```

right?

Reply:
SG.  I found a bug in Extract when expanding the test coverage. Fixing that too.  The example now covers this functions capabilities more extensively.